inwhale/core/quantizer.py

```
@@ -0,0 +1,35 @@
+    Base class for quantizers.
+
+    Quantization maps high precision floating-point numbers (say 32 bits) to low precision integers (say 8 bits). This reduces:
+    1. memory footprint (32 -> 8 bit = 75% reduction)
+    2. computational requirements (integer math is faster and less power-hungry than floating point math)
+
+    The challenge is doing this with minimal accuracy loss.
+    """
+
+    def __init__(self, bits: int):
+        if bits < 0:
+            raise ValueError("Bits must be positive.")
```

Ruff (0.14.8)

Avoid specifying long messages outside the exception class

(TRY003)